pkg/roachprod/vm/gce/utils.go

```diff
@@ -31,7 +31,11 @@ import (
 const gceDiskStartupScriptTemplate = `#!/usr/bin/env bash
 # Script for setting up a GCE machine for roachprod use.
 
-set -x
+# ensure any failure fails the entire script
+set -eux
+
+# Redirect output to a log file
+exec > /var/log/roachprod_startup.log 2>&1
```

Comment:
Related to @nameisbhaskar's comment: I think it would be nice to log both to the file _and_ to stdout/err as before. The advantage is that then the output of the startup script will be available on google cloud's logs which is useful.

I believe you can achieve that with something like:

```bash
set -eux

exec &> >(tee {{ .StartupLogs }})

# rest of the script
```

Reply:
Updated